pennylane/ops/functions/equal.py

```diff
@@ -99,3 +124,35 @@ def equal(
                 return False
 
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
+
+
+def equal_measurements(op1, op2):
+    """Determine whether two MeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    if op1.obs is not None and op2.obs is not None:
+        observables_match = equal(op1.obs, op2.obs)
+    # check obs equality when either one is None (False) or both are None (True)
+    else:
+        observables_match = op1.obs == op2.obs
```

Comment:
I was torn on how to structure this conditional. 
Your suggestion `qml.equal(op1.obs, op2.obs) if op1.obs is not None else op1.obs == op2.obs` would have worked, but specifying `if op1.obs is not None` means when `op1.obs is None` and `op2.obs is not None` would be treated differently than `op1.obs is not None` and `op2.obs is None`.

The only time when we need to use `equal()` is when both are `not None`. If this structuring and the comment I left on line 134 would cause confusion, please let me know.

Reply:
Nice catch! This is definitely better 👍